File: src/main/java/org/checkerframework/checker/mungo/typestate/graph/exceptions/StateNotDefined.java
package org.checkerframework.checker.mungo.typestate.graph.exceptions;

import org.checkerframework.checker.mungo.typestate.ast.TIdNode;

public class StateNotDefined extends TypestateError {

  public static final long serialVersionUID = 0L;

  public final TIdNode id;

  public StateNotDefined(TIdNode id) {
    this.id = id;
  }

  @Override
  public String getMessage() {
    return String.format("State %s was not defined (%s)", id.name, id.pos.toString());
  }

}

File: src/main/kotlin/org/checkerframework/checker/mungo/typestate/graph/exceptions/StateNotDefined.kt
package org.checkerframework.checker.mungo.typestate.graph.exceptions

import org.checkerframework.checker.mungo.typestate.ast.TIdNode

class StateNotDefined(val id: TIdNode) : TypestateError() {
  override val message: String
    get() = String.format("State %s was not defined (%s)", id.name, id.pos.toString())

  companion object {
    const val serialVersionUID = 0L
  }
}
